sky/exceptions.py

```diff
@@ -29,16 +33,28 @@ def with_failover_history(self, failover_history: List[Exception]) -> None:
         return self
 
 
-class SpotJobFailedBeforeProvisionError(Exception):
+class ProvisionPrechecksError(Exception):
     """Raised when a spot job fails before provision.
 
+    Please refer to the docstring of `spot.recovery_strategy._launch` for
```

Comment:
```suggestion
    Developer note: For now this should only be used by managed spot code path (technically, this can/should be raised by the lower-level sky.launch()). Please refer to the docstring of `spot.recovery_strategy._launch` for
```